pulsar/consumer_partition.go

```diff
@@ -949,6 +949,11 @@ func (pc *partitionConsumer) Close() {
 
 	// wait for request to finish
 	<-req.doneCh
+
+	// It will close `queueCh.in`. If `MessageReceived` was called after that, it will panic because new messages
+	// will be sent to a closed channel. However, generally it's impossible because the broker will not be able to
+	// dispatch messages to this consumer after receiving the close request.
+	pc.queueCh.stop()
```

Comment:
I see that you're not closing `closeCh` here but you're just closing `inCh` and `outCh`.

Beware because `closeCh` is an unbuffered channel and once you call `stop` the first time the listener of that channel will return so any subsequent `c.closeCh <- struct{}{}` is going to block forever and it will be hard to detect.

Wouldn't it be better to just close `closeCh`? It will have the same side effect but at least if you `stop` twice, you'll catch it instead of silently being stuck. Given that you're not planning to call `stop` multiple times, wouldn't it be better to close the channel then? Wdyt?